### PR TITLE
added timeout options to adhoc and console

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

adhoc, console

### CODE REVIEW

File: lib/ansible/cli/arguments/option_helpers.py [hunk 1]
```diff
@@ -351,6 +351,12 @@ def add_runtask_options(parser):
                         help="set additional variables as key=value or YAML/JSON, if filename prepend with @", default=[])
 
 
+def add_tasknoplay_options(parser):
+    """Add options for commands that run a task w/o a defined play"""
+    parser.add_argument('--task-timeout', dest="task_timeout", action="store", default=None,
```
Comment (Shrews) [inline lib/ansible/cli/arguments/option_helpers.py]: We say the default is `0` in several spots, yet here we use `None`. It works, just seems odd to not make it `0` for consistency sake.

File: test/units/cli/test_adhoc.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def test_play_ds_positive():
     adhoc_cli.parse()
     ret = adhoc_cli._play_ds('command', 10, 2)
     assert ret['name'] == 'Ansible Ad-Hoc'
-    assert ret['tasks'] == [{'action': {'module': 'command', 'args': {}}, 'async_val': 10, 'poll': 2}]
+    assert ret['tasks'] == [{'action': {'module': 'command', 'args': {}}, 'async_val': 10, 'poll': 2, 'timeout': None}]
```
Comment (s-hertel) [inline test/units/cli/test_adhoc.py]: s/None/0/

File: lib/ansible/cli/console.py [hunk 3]
```diff
@@ -91,6 +92,8 @@ def init_parser(self):
         opt_help.add_fork_options(self.parser)
         opt_help.add_module_options(self.parser)
         opt_help.add_basedir_options(self.parser)
+        opt_help.add_runtask_options(self.parser)
```
Comment (s-hertel) [inline lib/ansible/cli/console.py:95]: Extra vars could have a minor_changes entry too.

**Review by s-hertel — APPROVED** [review]

File: lib/ansible/cli/console.py [hunk 4]
```diff
@@ -354,6 +361,20 @@ def do_diff(self, arg):
...
+    def do_timeout(self, arg):
+        """Set the timeout"""
+        if arg:
+            try:
+                timeout = int(arg)
+                if timeout < 0:
+                    display.error('The timeout must be greater than or equal to 1, use 0 to disable')
+                else:
+                    self.task_timeout = timeout
+            except TypeError as e:
```
Comment (s-hertel) [inline lib/ansible/cli/console.py]: Maybe this should handle ValueError too.

**Pull request merged**